compose.yml

```diff
@@ -285,6 +289,38 @@ services:
 
   sqlite:
     <<: *base
+    image: django-docker-box:${PYTHON_IMPLEMENTATION}-${PYTHON_VERSION}-sqlite${SQLITE_VERSION}
+    pull_policy: never
+    build:
+      context: .
+      dockerfile_inline: |
+        FROM django-docker-box:${PYTHON_IMPLEMENTATION}-${PYTHON_VERSION}
+        SHELL ["/bin/bash", "-o", "errexit", "-o", "nounset", "-o", "pipefail", "-o", "xtrace", "-c"]
+        # Only compile SQLite and set LD_PRELOAD if a version is specified.
+        RUN <<EOF
+          if [[ "${SQLITE_VERSION}" ]]; then
+            export CFLAGS="${SQLITE_CFLAGS}"
+            git clone --depth 1 --branch version-${SQLITE_VERSION} \
+            https://github.com/sqlite/sqlite.git /tmp/sqlite
+            cd /tmp/sqlite
+            ./configure
+            make
+            if [ -f libsqlite3.so ]; then
+              cp libsqlite3.so /tmp/
+            else
+              cp .libs/libsqlite3.so /tmp/
+            fi
+            rm -rf /tmp/sqlite
```

Comment:
The "configure" script underwent major refactoring in 3.48.0: https://www.sqlite.org/releaselog/3_48_0.html

As a result, the compiled library is placed directly in the source root. In older versions, it's placed inside a `.libs` directory.

The `rm -rf` is just a cleanup step to keep the image size small.